action/protocol/rewarding/protocol.go

```diff
@@ -184,9 +184,6 @@ func createGrantRewardAction(rewardType int, height uint64) action.Envelope {
 
 // Validate validates a reward action
 func (p *Protocol) Validate(ctx context.Context, act action.Action, sr protocol.StateReader) error {
-	if !protocol.MustGetFeatureCtx(ctx).ValidateRewardProtocol {
-		return nil
-	}
```

Comment:
Doesn't this cause the fullsync to fail for heights before Newfoundland?

Reply:
no, it still pass, for all blocks before Newfoundland, all the reward actions are also generated by block producer, i.e., all delegates are honest, no one is doing bad things.

The flag is added for safety (skip the check for blocks before hard-fork), after the hard-fork, we can verify that blocks before it also meet the added check, so the flag can be removed. This technique has been used multiple times in the past.